### PR TITLE
Support for btn-spin for remix icons

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Seems like we start using remix icon https://github.com/mautic/mautic/pull/13318
This PR add support for remix icons for btn-spinner

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Onces If we add new icons, it should be tested and merged

### CODE REVIEW

**Review by LordRembo — CHANGES_REQUESTED** [review]
To make sure this doesn't accidentally target 'normal' classes that start with `ri`, I would suggest using `ri-` instead.

**Comment by kuzmany** [comment]
@LordRembo good point. I've added it.

**Comment by LordRembo** [comment]
Thinking a bit further, I think it doesn't really make much sense to add the 'spin' functionality like this, because the remix icons will eventually replace the FontAwesome icons. At which point, the `fa-spin` class won't work anymore.
It would be best to either use a class/function provided by the remix iconpack itself, or make a custom one.
@andersonjeccel Does the remix iconpack have a class or function that is equivalent to FontAwesome's spinning functionality?

**Comment by LordRembo** [comment]
@andersonjeccel maybe you can suggest a good way to add this instead of relying on the FontAwesome classes.

File: app/bundles/CoreBundle/Assets/js/1.core.js [hunk 1]
```diff
@@ -407,7 +407,8 @@ var Mautic = {
 
         if (mQuery(target).length) {
             var hasBtn = mQuery(target).hasClass('btn');
-            var hasIcon = mQuery(target).hasClass('fa');
+            var className = mQuery(target).attr('class');
+            var hasIcon = mQuery(target).hasClass('fa') || className.startsWith('ri-');
```
Comment (andersonjeccel) [inline app/bundles/CoreBundle/Assets/js/1.core.js:411]: var hasIcon = mQuery(target).hasClass('fa') || className.startsWith('ri-loader');

**Review by andersonjeccel — CHANGES_REQUESTED** [review]
Improved class to prevent conflicts

@LordRembo

**Comment by andersonjeccel** [comment]
@LordRembo Remix does not provide a function for spinning

**Review by LordRembo — CHANGES_REQUESTED** [review]
Ah, alright, then I propose this PR copies the 'spinning' css behind FontAwesome and puts that in the CSS files. Basically following the same logic.
I believe there's keyframe animations that are used by the `fa-spinner` and/or  `fa-spin` classes that should 1) be copied and the used on a generic `ri-spinner` class.
And 2), the `ri-spinner` class should then be added to the element with `ri-loader` (or whatever icon it is that needs to be spinning).

**Comment by kuzmany** [comment]
I am a little lost. Anyway, anybody can pick it up and update it to the way of new design patterns. We do not forget also we use in some cases .btn-nospin to prevent use of spinner if we do not need it. Also, you can decide not to use spinner anymore for new design patterns, if not needed.

The goal of the PR was to bring spinner support during the state while we use fa- and ri- icons. 
After removing FA icons, it's obsolete.

**Review by LordRembo — APPROVED** [review]
@kuzmany Yeah, you're right. Sorry, I'm thinking a bit too far ahead. Perfectly fine to make this change now and revamp later.

**Comment by LordRembo** [comment]
@andersonjeccel Maybe you can mark as 'approved' too, then this thing is ready to push forward

**Review by andersonjeccel — APPROVED** [review]

**Comment by kuzmany** [comment]
Reopen once If need it ;)

**Comment by andersonjeccel** [comment]
@kuzmany Could you implement that change I suggested for the icon class?
It needs to start with ri-loader to avoid conflicts with other icons
Then we're good to go

This feature is necessary, thanks for implementing it

**Comment by kuzmany** [comment]
@andersonjeccel Your change does not make sense. This condition checks if elements (button or link) have an icon. If yes, then switch to spinner; after finishing request, get it back.

**Comment by andersonjeccel** [comment]
@kuzmany 

ri-loader is a prefix for loading icons

we need to make sure the spin feature will apply only to these icons, I suppose 

image

how could we fix this? 
I'll leave it open for you to code

thanks again for catching this issue 🙏🏻

**Comment by kuzmany** [comment]
Outdated

**Pull request closed without merging**